### PR TITLE
fix/breaking tests

### PR DESCRIPTION
**Pull request by EduardoNeville**
In this PR we will fix the issues from the previous merged pr. This includes
a navigation commit that was not implemented as well as the removal of tests 
currently not working.

### CODE REVIEW

File: __tests__/e2e-tests/ForgottenPassword-test.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}));
+
+const SCREEN_WIDTH = Dimensions.get("window").width;
+
+describe("MyScreen Component", () => {
+  const mockRouter = {
+    push: useRouter().navigate,
+  };
+
+  it("navigue vers un autre screen quand le bouton est pressé", () => {
```
Comment (margdsq) [inline __tests__/e2e-tests/ForgottenPassword-test.tsx:20]: Can you change the language to English ? For unity with other tests

File: __tests__/e2e-tests/ForgottenPassword-test.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          height: 0,
+        },
+        layoutMeasurement: {
+          width: SCREEN_WIDTH, // Largeur de la zone visible
+          height: 0,
+        },
+      },
+    });
+
+    // Vérifiez que la page finale est affichée
```
Comment (margdsq) [inline __tests__/e2e-tests/ForgottenPassword-test.tsx:44]: Same for the comments

**Review by margdsq — CHANGES_REQUESTED** [review]

File: __tests__/UI-tests/Forgot-Password-test.tsx
Comment (Roman-batut) [inline __tests__/UI-tests/Forgot-Password-test.tsx:1]: I think we should be working with testIDs but as this is managed another PR we should just be careful when merging that we overwrite the correct implementation of these tests.

File: __tests__/Maximise-Screen-test.tsx
Comment (Roman-batut) [inline __tests__/Maximise-Screen-test.tsx:1]: I see we leave these tests as always passing for the purpose of what the scope of this PR is, this scope is indeed managed by another PR let's just be careful when we merge both PRs to implement the correct tests.

File: app/screens/auth/set_up_screen.tsx [hunk 3]
```diff
@@ -9,6 +9,7 @@ import { ThemedText } from "@/components/theme/ThemedText";
 import { ThemedScrollView } from "@/components/theme/ThemedScrollView";
 import { TouchableOpacity, Image } from "react-native";
 import * as ImagePicker from "expo-image-picker";
+import { useRouter } from "expo-router";
```
Comment (Roman-batut) [inline app/screens/auth/set_up_screen.tsx:12]: This isn't used as the navigation system has been updated, unsure if this is the case in master as well. Either way we should remove it or make sure it doesn't merge into master the way it is right now.

File: app/screens/home/home_screen.tsx [hunk 4]
```diff
@@ -6,34 +6,19 @@ import { ThemedScrollView } from "@/components/theme/ThemedScrollView";
 import { ThemedView } from "@/components/theme/ThemedView";
 import { BottomBar } from "@/components/navigation/BottomBar";
 import { auth } from "@/firebase/Firebase"; // Import auth
-import { DBChallenge } from "@/firebase/FirestoreCtrl";
+import getUser, { DBChallenge } from "@/firebase/FirestoreCtrl";
```
Comment (Roman-batut) [inline app/screens/home/home_screen.tsx:9]: getUser isn't used in the file neither is auth

File: app/screens/home/maximize_screen.tsx [hunk 5]
```diff
@@ -8,6 +8,7 @@ import { SingleComment, CommentType } from "@/components/posts/Comment";
 import { ThemedScrollView } from "@/components/theme/ThemedScrollView";
 import { getAuth } from "firebase/auth";
 import { ThemedTextInput } from "@/components/theme/ThemedTextInput";
+import { useRouter } from "expo-router";
```
Comment (Roman-batut) [inline app/screens/home/maximize_screen.tsx:11]: Same as set-up screen

File: app/screens/auth/forgot_password_screen.tsx [hunk 6]
```diff
@@ -3,6 +3,7 @@ import { ThemedView } from "@/components/theme/ThemedView";
 import { ThemedText } from "@/components/theme/ThemedText";
 import { ThemedTextInput } from "@/components/theme/ThemedTextInput";
 import { ThemedTextButton } from "@/components/theme/ThemedTextButton";
+import { useRouter } from "expo-router";
```
Comment (Roman-batut) [inline app/screens/auth/forgot_password_screen.tsx:6]: Same as set-up screen

File: app/screens/welcome/final_screen.tsx [hunk 7]
```diff
@@ -3,14 +3,15 @@ import { ThemedTextButton } from "@/components/theme/ThemedTextButton";
 import { ThemedView } from "@/components/theme/ThemedView";
 import React from "react";
 import { StyleSheet, Dimensions } from "react-native";
+import { useRouter } from "expo-router";
```
Comment (Roman-batut) [inline app/screens/welcome/final_screen.tsx:6]: Same as set-up screen.

File: navigation/Navigation.tsx
Comment (Roman-batut) [inline navigation/Navigation.tsx:1]: Nice catch, indeed passing firestoreCtrl as a function was the problem with it not working in any other files using it.

File: navigation/Navigation.tsx [hunk 8]
```diff
@@ -44,34 +44,34 @@ export const Nav: React.FC<AppStackProps> = ({
...
         <Screen name="ForgotPassword">
           {(props: any) => (
             <ForgotPasswordScreen
               {...props}
-              firestoreCtrl={{ firestoreCtrl }}
+              firestoreCtrl={ firestoreCtrl }
             />
           )}
         </Screen>
         <Screen name="SetUp">
```
Comment (Roman-batut) [inline navigation/Navigation.tsx:69]: Here the sign-up screen is searching for a navigation named SetUser when trying to set up an account in the only button attached to this screen. Resulting in an error and we can't go further than sign-up screen as a new user. Therefore we should modify the Screen name in this file to SetUser or modify the action made by the button to reflect this name. 

```suggestion
        <Screen name="SetUser">
```

Here's the error when trying to sign-up : 

```
(NOBRIDGE) ERROR  The action 'NAVIGATE' with payload {"name":"SetUser"} was not handled by any navigator.

Do you have a route named 'SetUser'?
```

**Review by Roman-batut — CHANGES_REQUESTED** [review]
### Summary :
Overall the PR fixes the breaking tests we had, there are small adjustments to make as well as bigger ones that I am unsure are in the scope of this PR although it's always better to keep these in mind for future PRs or even when merging this into master.

### Requested Changes :
- Minor ones include unused imports and keeping in mind some aspects of our code while merging. Also this version hasn't been prettified yet.

- More important ones but may be out of the scope of this PR would be fixing the navigation between sign-up and set-up screens. 
As well as an issue happening when posting a challenge with the firebase :
```
(NOBRIDGE) ERROR  Error writting challenge document:  [FirebaseError: Function addDoc() called with invalid data. Unsupported field value: undefined (found in field uid in document challenges/5o37UHMtbldn9TNDRsTO)]
(NOBRIDGE) ERROR  Error creating challenge:  [FirebaseError: Function addDoc() called with invalid data. Unsupported field value: undefined (found in field uid in document challenges/5o37UHMtbldn9TNDRsTO)]
```

**Review by Roman-batut — APPROVED** [review]
Changes reflects comments previously made.

### LTGM

**Review by margdsq — APPROVED** [review]

**Pull request merged**